server/broadcast_test.go

```diff
@@ -260,13 +287,11 @@ func TestSelectSession(t *testing.T) {
 	assert.Len(bsm.sessList(), 0)
 	assert.Len(bsm.sessMap, 2) // map should still track original sessions
 
-	// assert session list gets refreshed if under threshold. check via waitgroup
+	// assert session list gets refreshed if under threshold
 	bsm.numOrchs = 1
-	var wg sync.WaitGroup
-	wg.Add(1)
-	bsm.createSessions = func() ([]*BroadcastSession, error) { wg.Done(); return nil, fmt.Errorf("err") }
-	bsm.selectSession()
-	assert.True(wgWait(&wg), "Session refresh timed out")
+	updateBsmSessList(bsm.BroadcastSessionsManager, []*BroadcastSession{StubBroadcastSession("transcoder3")})
+	sess = bsm.selectSession()
+	assert.Equal(bsm.sessMap["transcoder3"], sess)
```

Comment:
I don't think this updated test is checking the same thing as the old test.

The old test used a wait group to make sure that `createSessions()` was called asynchronously by `selectSession()`. Looks like this updated test adds transcoder3 to the sessMap using `updateBsmSessList()` and then checks that sessMap contains transcoder3 which would be checking the behavior of `updateBsmSessList()` as opposed to `selectSession()`.

Reply:
fixed in 2c78fb9a784ea81c42a9bdbd0d91c71ab6fc255b